src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DatabaseHandler.java

```diff
@@ -0,0 +1,273 @@
+        String eventID = "";
+
+        String sqlSelectStatement = String.format("SELECT %s FROM %s WHERE %s=?", EVENT_ID_KEY, table, table.keyName);
+        try (Connection connection = connect();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlSelectStatement)) {
+
+            preparedStatement.setString(1, searchCriteria);
+            eventID = executeQuery(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to fetch values from database: {}\n{}", e.getMessage(), e);
+        }
+
+        if (eventID.isEmpty()) {
+            throw new NoSuchElementException("Database did not return any value for this query. Table: "
+                    + table + ",KeyName: " + table.keyName + ",searchCriteria: " + searchCriteria);
+        }
+
+        return eventID;
+    }
+
+    /**
+     * This function updates value to the given table. The searchCriteria value is
+     * different depending on Table (branch name for scs and change-id for scc)
+     *
+     * @param table
+     * @param searchCriteria
+     * @param eiffelEvent
+     * @throws ConnectException
+     * @throws SQLException
+     */
+    public void updateInto(final Table table, final String searchCriteria, final String eiffelEvent)
+            throws ConnectException, SQLException {
+        String sqlUpdateStatement = String.format("UPDATE %s SET %s=? WHERE %s=?", table, EVENT_ID_KEY, table.keyName);
+        prepareAndExecuteStatement(sqlUpdateStatement, searchCriteria, eiffelEvent);
+    }
+
+    /**
+     * This function inserts values to the given table. The key value is
+     * different depending on Table (branch name for scs and change-id for scc)
+     *
+     * @param table
+     * @param key
+     * @param value
+     * @throws SQLException
+     * @throws ConnectException
+     */
+    public void insertInto(final Table table, final String key, final String value)
+            throws SQLException, ConnectException {
+        String sqlInsertStatement = String.format("INSERT INTO %s(%s,%s) VALUES(?,?)", table, EVENT_ID_KEY,
+                table.keyName);
+        prepareAndExecuteStatement(sqlInsertStatement, key, value);
+
+    }
+
+    /**
+     * This function will create a sql database structure in the database file if it
+     * does not exist.
+     *
+     * @throws ConnectException
+     */
+    private void createNewDatabase() throws ConnectException {
+        try (Connection sqlConnection = connect()) {
+            if (sqlConnection != null) {
+                String parentPath = buildParentFilePath();
+                createParentDirsIfNecessary(parentPath);
+
+                //When getMetaData is called, it will create the .db file
+                sqlConnection.getMetaData();
+                sqlConnection.close();
+            }
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to create new database: {}\n{}", e.getMessage(), e);
```

Comment:
I believe e contains the message if printed. You don't need to add it with brackets either. just adding e as a parameter is enough to have in printed. Ex: ("Error when trying to create new database.", e)

Reply:
Agreed :)